py_ecc/fields/field_elements.py

```diff
@@ -349,11 +366,11 @@ class FQ2(FQP):
     The quadratic extension field
     """
 
-    degree = 2
-    FQ2_MODULUS_COEFFS = None  # type: FQ2_modulus_coeffs_type
+    degree: int = 2
+    FQ2_MODULUS_COEFFS: "FQ2_modulus_coeffs_type"
 
     def __init__(self, coeffs: Sequence[IntOrFQ]) -> None:
-        if self.FQ2_MODULUS_COEFFS is None:
+        if not hasattr(self, "FQ2_MODULUS_COEFFS"):
```

Comment:
Should this allow a value of `None` if the attribute is present? I think the `is None` check is still needed.

Another option is to first assign a variable like `x = self.get("FQ2_MODULUS_COEFFS")` and then use the same logic as before.

Reply:
This changed along with changing how the attributes of these classes were defined. Previously, it was:

`FQ12_MODULUS_COEFFS = None  # type: FQ12_modulus_coeffs_type`

This then makes it either the `FQ12_modulus_coeffs_type` or `None`, which then required a ton of addition validation through implementations to make sure that it's not `None` whenever it's used. If it's just declared without assignment, then it's there, and typed, ready for subclasses to assign a value to it without requiring the extra `if not None` nonsense.